### PR TITLE
Add loki and db layer

### PR DESCRIPTION
**Pull request by joybro**
loki is lightweight memory nosql. This will be used for development
when this project is running on local. In Lambda, DynamoDB will be
used instead. And these different db will be abstracted by db layer.

### CODE REVIEW

File: server/db/loki.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+const updateArticle = (id: string, content: string, currentRev: number): boolean => {
+  const article = articles.findOne({ id });
+
+  assert(currentRev >= 0);
+
+  if (article.rev !== currentRev) return false;
+
+  article.content = content;
+  article.rev++;
```
Comment (bongsok) [inline server/db/loki.js]: Are these two lines above needed? They look like duplicated with return statement.

Comment (joybro) [reply server/db/loki.js]: nice catch!

**Comment by joybro** [comment]
@bongsok uploaded 2nd revision.

**Comment by bongsok** [comment]
Great.

**Pull request merged**